src/urls/url-handler.ts

```diff
@@ -33,4 +33,9 @@ export interface UrlHandler {
       string;
   convertUrl(url: OriginalDocumentUrl): ConvertedDocumentUrl;
   createConvertedUrl(partialUrl: string): ConvertedDocumentUrl;
+
+  originalUrlToPackageRelative(url: OriginalDocumentUrl): string;
+  convertedUrlToPackageRelative(url: ConvertedDocumentUrl): string;
+  convertedDocumentFilePathToPackageRelative(url: ConvertedDocumentFilePath):
```

Comment:
The current system of project-specific URL handling is definitely hitting a breaking point here, we keep needing to add one-off workarounds to each specific URL handler.

I think we have agreement about *wanting* to move to a single workspace-mode conversion. I'm going to experiment with how easy it would be to preserve single-package conversion inside that layout, by doing something like cloning the local git repo at it's current branch into the workspace.